code/modules/projectiles/guns/energy.dm

```diff
@@ -666,35 +739,61 @@
 
 	aim_slowdown = 0.7
 	wield_delay = 0.7 SECONDS
-	scatter = 0
+	scatter = -4
 	scatter_unwielded = 10
-	fire_delay = 1 SECONDS
+	fire_delay = 0.8 SECONDS
 	accuracy_mult = 1.2
 	accuracy_mult_unwielded = 0.5
 	movement_acc_penalty_mult = 6
 	mode_list = list(
 		"Standard" = /datum/lasrifle/base/energy_sniper_mode/standard,
 		"Heat" = /datum/lasrifle/base/energy_sniper_mode/heat,
+		"Shatter" = /datum/lasrifle/base/energy_sniper_mode/shatter,
+		"Ricochet" = /datum/lasrifle/base/energy_sniper_mode/ricochet,
 	)
 
 /datum/lasrifle/base/energy_sniper_mode/standard
-	rounds_per_shot = 50
-	fire_delay = 1 SECONDS
+	rounds_per_shot = 30
+	fire_delay = 0.8 SECONDS
 	ammo_datum_type = /datum/ammo/energy/lasgun/marine/sniper
 	fire_sound = 'sound/weapons/guns/fire/Laser Sniper Standard.ogg'
 	message_to_user = "You set the sniper rifle's charge mode to standard fire."
 	fire_mode = GUN_FIREMODE_SEMIAUTO
 	icon_state = "tes"
+	description = "Fires a single strong laser pulse, with good damage and penetration, and no falloff."
 
 /datum/lasrifle/base/energy_sniper_mode/heat
-	rounds_per_shot = 150
+	rounds_per_shot = 100
 	fire_delay = 1 SECONDS
 	ammo_datum_type = /datum/ammo/energy/lasgun/marine/sniper_heat
 	fire_sound = 'sound/weapons/guns/fire/laser3.ogg'
 	message_to_user = "You set the sniper rifle's charge mode to wave heat."
 	fire_mode = GUN_FIREMODE_SEMIAUTO
 	icon_state = "tes"
 	radial_icon_state = "laser_heat"
+	description = "Fires an incendiary laser pulse, designed to ignite victims at range."
+
+/datum/lasrifle/base/energy_sniper_mode/shatter
+	rounds_per_shot = 100
+	fire_delay = 1 SECONDS
+	ammo_datum_type = /datum/ammo/energy/lasgun/marine/shatter
+	fire_sound = 'sound/weapons/guns/fire/laser_rifle_2.ogg'
+	message_to_user = "You set the sniper rifle's charge mode to shatter."
+	fire_mode = GUN_FIREMODE_SEMIAUTO
+	icon_state = "tes"
+	radial_icon_state = "laser_charge"
+	description = "Fires a devestating laser pulse that significantly degrades the victims armor, at the cost of lower direct damage."
+
+/datum/lasrifle/base/energy_sniper_mode/ricochet
+	rounds_per_shot = 45
+	fire_delay = 0.8 SECONDS
+	ammo_datum_type = /datum/ammo/energy/lasgun/marine/ricochet/four
```

Comment:
Starting at four and going down is still weird and I don't like it.

Reply:
same reason the other bouncy ammo does it - inherited bonus ammo.